kotlin/samples/containers/android/src/main/java/com/squareup/sample/container/masterdetail/MasterDetailContainer.kt

```diff
@@ -53,34 +52,46 @@ class MasterDetailContainer(
     }
   }
 
-  override fun showRendering(rendering: MasterDetailScreen) {
-    if (singleStub == null) renderSplitView(rendering)
-    else renderSingleView(rendering, singleStub)
+  override fun showRendering(
+    rendering: MasterDetailScreen,
+    hints: Hints
+  ) {
+    if (singleStub == null) renderSplitView(rendering, hints)
+    else renderSingleView(rendering, hints, singleStub)
```

Comment:
Kind of unfortunate that the hints have to be manually plumbed through everything, flutter contexts/compose ambients are all implicit (idk about swiftUI's Environment).

Reply:
I suppose. I kind of like the lack of magic in this. And in use it's pretty clear:

 * Every related call requires a `Hints` parameter
 * With the exception of the `WorkflowLayout` implementation, every place you make such a call is in the context of a function body that received a `Hints` argument